median/plugin.go

```diff
@@ -65,3 +71,45 @@ func (r *reportingPluginFactoryService) Close() error {
 func (r *reportingPluginFactoryService) HealthReport() map[string]error {
 	return map[string]error{r.Name(): r.Healthy()}
 }
+
+// chainReaderContract adapts a [types.ChainReader] to [median.MedianContract].
+type chainReaderContract struct {
+	chainReader types.ChainReader
+	contract    types.BoundContract
+}
+
+type latestTransmissionDetailsResponse struct {
+	ConfigDigest    ocrtypes.ConfigDigest
+	Epoch           uint32
+	Round           uint8
+	LatestAnswer    *big.Int
+	LatestTimestamp time.Time
+}
+
+type latestRoundRequested struct {
+	ConfigDigest ocrtypes.ConfigDigest
+	Epoch        uint32
+	Round        uint8
+}
+
+func (c *chainReaderContract) LatestTransmissionDetails(ctx context.Context) (configDigest ocrtypes.ConfigDigest, epoch uint32, round uint8, latestAnswer *big.Int, latestTimestamp time.Time, err error) {
+	var resp latestTransmissionDetailsResponse
+
+	err = c.chainReader.GetLatestValue(ctx, c.contract, "LatestTransmissionDetails", nil, &resp)
+	if err != nil {
+		return
+	}
+
+	return resp.ConfigDigest, resp.Epoch, resp.Round, resp.LatestAnswer, resp.LatestTimestamp, nil
+}
+
+func (c *chainReaderContract) LatestRoundRequested(ctx context.Context, lookback time.Duration) (configDigest ocrtypes.ConfigDigest, epoch uint32, round uint8, err error) {
+	var resp latestRoundRequested
+
+	err = c.chainReader.GetLatestValue(ctx, c.contract, "LatestRoundReported", map[string]any{"lookback": lookback}, &resp)
```

Comment:
lookback is an obsolete parameter unused anywhere at Chainlink.  We can't support it with `GetLatestValue()` without changing the meaning of the method and breaking the chain agnostic nature of it in an awkward way. Only contract method params (or equivalently, log topics) should be passed in params field to `GetLatestValue()`